### PR TITLE
Ensure VP number, RNG type and RNG seed can be set in any order

### PR DESCRIPTION
**Pull request by heplesser**
This PR ensures that VP number, RNG type and RNG seed can be set in any order. Also converts ticket-754.sli to Pytest and extends it to catch previously uncaught cases.

### CODE REVIEW

**Review by nicolossus — APPROVED** [review]

File: testsuite/pytests/sli2py_regressions/ticket_754.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+@pytest.mark.skipif_missing_threads
+@pytest.mark.parametrize('proc_param_name', ['local_num_threads', 'total_num_virtual_procs'])
+class TestSeedingAndProcSetting:
+    """
+    Tests that setting seed and number of thread or of VPs works in arbitrary order.
+    """
+
+    NUM_PROCS = 2
+
```
Comment (jougs) [inline testsuite/pytests/sli2py_regressions/ticket_754.py:67]:
```suggestion
TEST_SEED = 123

```

File: testsuite/pytests/sli2py_regressions/ticket_754.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        nest.rng_seed = test_seed
+        assert nest.rng_seed == test_seed
+
+    def test_seed_then_set_procs(self, proc_param_name):
+        """
+        Test that seeding and then setting number of procs gives correct seed.
+        """
+
+        test_seed = 123
+        nest.rng_seed = test_seed
```
Comment (jougs) [inline testsuite/pytests/sli2py_regressions/ticket_754.py:92]:
```suggestion
        nest.rng_seed = TEST_SEED
```

File: testsuite/pytests/sli2py_regressions/ticket_754.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        nest.set(**{proc_param_name: self.NUM_PROCS})
+        assert nest.rng_seed == test_seed
```
Comment (jougs) [inline testsuite/pytests/sli2py_regressions/ticket_754.py:94]:
```suggestion
        assert nest.rng_seed == TEST_SEED
```

File: testsuite/pytests/sli2py_regressions/ticket_754.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+@pytest.mark.skipif_missing_threads
+@pytest.mark.parametrize('proc_param_name', ['local_num_threads', 'total_num_virtual_procs'])
+class TestRngTypeAndProcSetting:
+    """
+    Tests that setting rng type and number of thread or of VPs works in arbitrary order.
+    """
+
+    NUM_PROCS = 2
+
```
Comment (jougs) [inline testsuite/pytests/sli2py_regressions/ticket_754.py:114]: Do the same as for `test_seed`, but for `test_rng_type`

**Review by jougs — CHANGES_REQUESTED** [review]
Almost good. I just would make some variables class constants to have them defined centrally and unclutter the actual code

**Review by jougs — APPROVED** [review]
LGTM

**Pull request merged**